File: core/src/main/java/com/larryhsiao/nyx/core/attachments/WrappedAttachment.java
package com.larryhsiao.nyx.core.attachments;

/**
 * Wrapped class of attachment
 */
public class WrappedAttachment implements Attachment {
    private final Attachment origin;

    public WrappedAttachment(Attachment origin) {
        this.origin = origin;
    }

    @Override
    public long id() {
        return origin.id();
    }

    @Override
    public long jotId() {
        return origin.jotId();
    }

    @Override
    public String uri() {
        return origin.uri();
    }

    @Override
    public int version() {
        return origin.version();
    }

    @Override
    public boolean deleted() {
        return false;
    }
}

File: core/src/main/java/com/larryhsiao/nyx/core/attachments/WrappedAttachment.kt
package com.larryhsiao.nyx.core.attachments

/**
 * Wrapped class of attachment
 */
open class WrappedAttachment(private val origin: Attachment) : Attachment {
    override fun id(): Long {
        return origin.id()
    }

    override fun jotId(): Long {
        return origin.jotId()
    }

    override fun uri(): String {
        return origin.uri()
    }

    override fun version(): Int {
        return origin.version()
    }

    override fun deleted(): Boolean {
        return origin.deleted()
    }
}